internal/pkg/configmanager/pipelinetemplate.go

```diff
@@ -0,0 +1,89 @@
+package configmanager
+
+import (
+	"fmt"
+
+	"github.com/imdario/mergo"
+	"gopkg.in/yaml.v3"
+)
+
+type (
+	PipelineTemplate struct {
+		Name    string     `yaml:"name"`
+		Type    string     `yaml:"type"`
+		Options RawOptions `yaml:"options"`
+	}
+)
+
+func findTemplateByName(name string, templates []PipelineTemplate) *PipelineTemplate {
```

Comment:
How about getTemplateByName? "Find" sounds like it returns true if found, while "get" sounds like getting the template if found.